presto-tests/src/test/java/com/facebook/presto/memory/TestMemoryManager.java

```diff
@@ -175,51 +194,64 @@ private void waitForQueryToBeKilled(DistributedQueryRunner queryRunner)
+            throws Exception
+    {
+        // Reserve all the memory
+        QueryId fakeQueryId = new QueryId("fake");
+        for (TestingPrestoServer server : queryRunner.getServers()) {
+            List<MemoryPool> memoryPools = server.getLocalMemoryManager().getPools();
+            assertEquals(memoryPools.size(), 1, "Only general pool should exist");
+            assertTrue(memoryPools.get(0).tryReserve(fakeQueryId, "test", memoryPools.get(0).getMaxBytes()));
+        }
 
-            List<Future<?>> queryFutures = new ArrayList<>();
-            for (int i = 0; i < 2; i++) {
-                queryFutures.add(executor.submit(() -> queryRunner.execute("SELECT COUNT(*), clerk FROM orders GROUP BY clerk")));
-            }
+        List<Future<?>> queryFutures = new ArrayList<>();
+        for (int i = 0; i < 2; i++) {
+            queryFutures.add(executor.submit(() -> queryRunner.execute("SELECT COUNT(*), clerk FROM orders GROUP BY clerk")));
+        }
 
-            // Wait for one of the queries to die
-            waitForQueryToBeKilled(queryRunner);
-
-            // Reserved pool shouldn't exist on the workers and allocation should have been done in the general pool
-            for (TestingPrestoServer server : queryRunner.getServers()) {
-                Optional<MemoryPool> reserved = server.getLocalMemoryManager().getReservedPool();
-                MemoryPool general = server.getLocalMemoryManager().getGeneralPool();
-                assertFalse(reserved.isPresent());
-                assertTrue(general.getReservedBytes() > 0);
-                // Free up the entire pool
-                general.free(fakeQueryId, "test", general.getMaxBytes());
-                assertTrue(general.getFreeBytes() > 0);
-            }
+        // Wait for one of the queries to die
+        waitForQueryToBeKilled(queryRunner);
+
+        // Reserved pool shouldn't exist on the workers and allocation should have been done in the general pool
+        for (TestingPrestoServer server : queryRunner.getServers()) {
+            Optional<MemoryPool> reserved = server.getLocalMemoryManager().getReservedPool();
+            MemoryPool general = server.getLocalMemoryManager().getGeneralPool();
+            assertFalse(reserved.isPresent());
+            assertTrue(general.getReservedBytes() > 0);
+            // Free up the entire pool
+            general.free(fakeQueryId, "test", general.getMaxBytes());
+            assertTrue(general.getFreeBytes() > 0);
+        }
 
-            for (Future<?> query : queryFutures) {
-                query.get();
-            }
+        for (Future<?> query : queryFutures) {
+            query.get();
         }
     }
 
-    @Test(timeOut = 240_000)
+    @BeforeGroups(groups = "testNoLeak")
+    public void testNoLeakSetup()
+            throws Exception
+    {
+        Map<String, String> properties = ImmutableMap.<String, String>builder()
+                .put("task.verbose-stats", "true")
```

Comment:
cluster pools setup also only sets task.verbose-stats can they share a group/query runner?  Also, looks like that config has changed its name to `task.per-operator-cpu-timer-enabled`. Can you update that here while you're doing this?